configs/experiment/bool_mask_audio_jepa.yaml

```
@@ -0,0 +1,26 @@
+# @package _global_
+
+defaults:
+  - override /models: bool_mask_audio_jepa
+  - override /trainers: bool_mask_audio_jepa
+
+interaction:
```

ObservationGeneratorのshapeを変える必要がある。
```suggestion
interaction:
  observation_generator:
    _args_:
      - ${shared.audio_channels}
      - ${shared.audio_sample_size}
```

---

ごめんこのsuggestionについて質問したい
結論から書くと、多分下のこれをどこか適切な箇所に移動するべきなんだろうけど、どこにどう記述していいかがわからない
```
_args_:
  - ${shared.audio_channels}
  - ${shared.audio_sample_size}
```
というのも、
1. `interaction`って直接`observation_generator`じゃなく、一度`environment`としてラップされたものを取るってふうに理解してるんだけど、このsuggestionだと直接`observation_generator`を取ってて、これは多分ミス？
2. `observation_generator`って`AudioFilesObservationGenerator`だけど、[__init__の引数](https://github.com/MLShukai/ami/blob/hifigan_configs/ami/interactions/environments/audio_observation_generator.py#L134-L138)を見ると`${shared.audio_channels}`を指定できるものがない

っていう2つから、もしかするとこれって本当は別な箇所に指定するものなんじゃないかと勝手に推測した

---

`experiment/bool_mask_audio_jepa.yaml`では `environment`を overrideしていないので、 [デフォルトのenvironment configファイル](https://github.com/MLShukai/ami/blob/main/configs/interaction/environment/dummy_image_io.yaml)を読み込む。
ここの `torch.zeros`に`_args_`のリストが上書きされて渡されるんだ。

もしこれがわかりにくければ、`configs/environment/dummy_audio_io.yaml`を追加して、`experiment/bool_mask_audio_jepa.yaml`のenvironment overrideを追加すると良い。

* `configs/environment/dummy_audio_io.yaml`
```yaml
_target_: ami.interactions.environments.dummy_environment.DummyEnvironment
observation_generator:
  _partial_: true
  _target_: torch.zeros
  _args_:
    - ${shared.audio_channels}
    - ${shared.audio_sample_size}

action_checker:
  _target_: ami.interactions.environments.dummy_environment.ActionTypeChecker
  action_type:
    _target_: hydra.utils.get_class
    path: torch.Tensor
```

* `configs/experiment/bool_mask_audio_jepa.yaml`
```yaml
defaults:
  - override /interaction/environment: dummy_audio_io
```

---

ありがとう助かる〜〜　dummy_audio_io追加した https://github.com/MLShukai/ami/pull/267/commits/2871e988673d990b211475c6fbda54e910471fde